tests/reftests/gen.ml

```diff
@@ -95,15 +95,31 @@ let () =
   in
   let process archive_hashes filename =
     let base_name = OpamStd.String.remove_suffix ~suffix:".test" filename in
-    let condition = match Filename.extension base_name with
-      | "" -> ""
-      | os ->
-        Printf.sprintf "\n (enabled_if (= %%{os_type} %S))"
-          String.(capitalize_ascii (sub os 1 (length os - 1)))
-    in
     if base_name = filename then archive_hashes else
+      let archive_hash, tags =
+        match first_line_tags ~path:filename with
+        | [] -> "", []
+        | ar::_ as tags -> ar, tags
+      in
+      let os_condition = match Filename.extension base_name with
+        | "" -> ""
+        | os ->
+          Printf.sprintf "(= %%{os_type} %S)"
+            String.(capitalize_ascii (sub os 1 (length os - 1)))
+      in
+      let tags_conditions =
+        List.map (fun tag ->
+            Printf.sprintf
+              "(= %%{env:TEST%s=0} 1)"
+              tag)
+          tags
+      in
```

Comment:
I think it's best to make the set of tags limited to make it easier to maintain, agree on what tags we allow and for the free typo detection
```suggestion
        | ar::tags -> ar, tags
      in
      let os_condition = match Filename.extension base_name with
        | "" -> ""
        | os ->
          Printf.sprintf "(= %%{os_type} %S)"
            String.(capitalize_ascii (sub os 1 (length os - 1)))
      in
      let tags_conditions =
        List.map (function
            | ("NETWORK" | "SANDBOX") as tag ->
              Printf.sprintf
                "(= %%{env:TEST%s=0} 1)"
                tag
              | tag -> raise (Failure (Printf.sprintf "Tag '%s' unrecognized" tag)))
          (if String.equal ar null_hash then ar :: tags else tags)
      in
```

Reply:
I'm not opposed to this, however right now we can only choose specific tests with a _present_ tag. So we have to be wary of that if we want to maintain a proper tag list.

Also, repo hashes are also considered for filtering and those are kind of arbitrary in any case.

Reply:
> Also, repo hashes are also considered for filtering and those are kind of arbitrary in any case.

yeah, i removed that handling in my suggestion because i thought it wasn't useful and complicates the list of environment variable that affects opam

> So we have to be wary of that if we want to maintain a proper tag list.

i think this PR is good way to get the discussion started, but you're right, I've updated the suggestion to only handle `N0REP0`

Reply:
I feel we should not include repo hash to the tags in that case, as we won't have a reliable way to concile restrictions on both. Adding a first tag for the proposed use-case (that coincidentally appears along N0REP0) would be more orderly.

Reply:
As mentioned on slack i don't think we should have a "quick" tag, as maintaining it more work than it's worth and it's meaning can't really be defined either. What was bad about using N0REP0 like my code suggestion above?

Reply:
I'm not against using repo hashes as filters but I think it doesn't fit well with tag restrictions, because we won't apply those to repo hashes (at least not in a pretty way). And making N0REP0 the unique exception feels unsatisfactory.

We could constraint ourselves to using only well-defined tags, but in that case hashes don't really fit either, neither my initial goal of having an informal selection of tests for fast development feedback.